agent/structs/service_definition.go

```diff
@@ -28,12 +28,19 @@ func (s *ServiceDefinition) NodeService() *NodeService {
 	return ns
 }
 
-func (s *ServiceDefinition) CheckTypes() (checks CheckTypes) {
-	s.Checks = append(s.Checks, &s.Check)
+func (s *ServiceDefinition) CheckTypes() (checks CheckTypes, err error) {
+	if !s.Check.Empty() {
+		_, err := s.Check.Valid()
+		if err != nil {
+			return nil, err
+		}
+		s.Checks = append(s.Checks, &s.Check)
+	}
 	for _, check := range s.Checks {
-		if check.Valid() {
-			checks = append(checks, check)
+		_, err := check.Valid()
+		if err != nil {
+			return nil, err
 		}
 	}
-	return
+	return s.Checks, nil
```

Comment:
I thought we were going to refactor this to return a fresh slice, but never alter `s.Checks`.

Reply:
Maybe I misunderstood, I thought we wanted to stop doing that and maintain s.Checks correctly. Can fix it though.

Reply:
We wanted to eliminate that side-effect.